app/Http/Controllers/MonitorsController.php

```diff
@@ -94,7 +102,13 @@ public function update(MonitorRequest $request, Monitor $monitor)
             'url' => $validated['url'],
             'uptime_check_enabled' => $validated['monitorUptime'],
             'uptime_check_interval_in_minutes' => $validated['uptimeCheckInterval'],
+            'domain_check_enabled' => $validated['monitorDomain'],
         ]);
+
+        if ($monitor) {
+            $this->addOrUpdateDomainDetails($monitor);
```

Comment:
I initially thought, irrespective to the domain check in enabled or not, the domain details will be fetched and stored in the database. I was using the check to determine the triggering of notification.

But now, the Monitor will store the expiration date when a new one is created, but while updating a monitor, the lookup will only be done, in case the `previous check was disabled` and `during saving the monitor the check was enabled`.